src/test/java/com/exactpro/th2/sailfish/utils/ImplementationIFilterTest.java

```diff
@@ -171,10 +174,98 @@ void testFilterWithException(String first) {
+                .build();
+
+        ComparisonResult result = getResult(actual, filter);
+
+        Assertions.assertEquals(status, result.getResult("compareFilter").getStatus());
+    }
+
+    private static List<Arguments> twoTimeFilterOperationWithPrecision() {
+        return List.of(
+                Arguments.of("2007-12-03T10:15:30", "2007-12-03T10:15:35", createDuration(5), StatusType.PASSED),
+                Arguments.of("2007-12-03T10:15:30", "2007-12-03T10:15:36", createDuration(5), StatusType.FAILED),
+                Arguments.of("2007-12-03T10:15:30", "2007-12-03T10:15:35.100000000", createDuration(5, 100000000), StatusType.PASSED),
+                Arguments.of("2007-12-03T10:15:30", "2007-12-03T10:15:35.100000001", createDuration(5, 100000000), StatusType.FAILED)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("twoTimeFilterOperationWithPrecision")
+    void testTimeFilterOperationFilterWithPrecision(String filterValue, String messageValue, Duration precision, StatusType status) {
+        RootMessageFilter filter = RootMessageFilter.newBuilder()
+                .setMessageType(MESSAGE_TYPE)
+                .setMessageFilter(MessageFilter.newBuilder()
+                        .putFields("compareFilter", ValueFilter.newBuilder()
+                                .setSimpleFilter(filterValue)
+                                .setOperation(FilterOperation.EQ_TIME_PRECISION)
+                                .build())
+                        .build())
+                .setComparisonSettings(RootComparisonSettings.newBuilder()
+                        .setTimePrecision(precision)
+                        .build())
+                .build();
+
+        Message actual = createMessageBuilder(MESSAGE_TYPE)
+                .putFields("compareFilter", getSimpleValue(messageValue))
+                .build();
+
+        ComparisonResult result = getResult(actual, filter);
+
+        Assertions.assertEquals(status, result.getResult("compareFilter").getStatus());
+    }
+
     private ComparisonResult getResult(Message actual, RootMessageFilter filter) {
         MessageWrapper actualIMessage = converter.fromProtoMessage(actual, false);
-        IMessage filterIMessage = converter.fromProtoFilter(filter.getMessageFilter(), filter.getMessageType());
+        IMessage filterIMessage;
+        if (filter.hasComparisonSettings()) {
+            filterIMessage = converter.fromProtoFilter(
+                    filter.getMessageFilter(),
+                    RootComparisonSettingsUtils.convertToFilterSettings(filter.getComparisonSettings()),
+                    filter.getMessageType());
+        } else {
+            filterIMessage = converter.fromProtoFilter(filter.getMessageFilter(), filter.getMessageType());
+        } 
 
         return MessageComparator.compare(actualIMessage, filterIMessage, new ComparatorSettings());
     }
+    
+    private static Duration createDuration(long seconds, Integer nanos) {
```

Comment:
There is no reason to use boxed type for nanos part. The `Duration` has nanos as int type which has the default value equals zero. So, we can simply pass 0 for nanos